src/config/logger.ts

```diff
@@ -56,34 +38,38 @@ const errorSerializer = (err) => {
 	};
 };
 
-const hashSerializer = (data?: string): string | undefined => {
-	if (!data || !isString(data)) {
-		return undefined;
-	}
-	return createHashWithSharedSecret(data);
-};
-
 export const defaultLogLevel: LogLevel = process.env.LOG_LEVEL as LogLevel || "info";
 
+const loggerStream = (safe = true): Logger.Stream => ({
+	type: "raw",
+	stream: safe ? new SafeRawLogStream(FILTERING_FRONTEND_HTTP_LOGS_MIDDLEWARE_NAME) : new UnsafeRawLogStream(FILTERING_FRONTEND_HTTP_LOGS_MIDDLEWARE_NAME),
+	closeOnExit: false
+});
+
 // TODO Remove after upgrading Probot to the latest version (override logger via constructor instead)
 export const overrideProbotLoggingMethods = (probotLogger: Logger) => {
 	// Remove  Default Probot Logging Stream
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
 	(probotLogger as any).streams.pop();
 
 	// Replace with formatOut stream
-	probotLogger.addStream({
-		type: "stream",
-		stream: LOG_STREAM,
-		closeOnExit: false,
-		level: defaultLogLevel
-	});
+	probotLogger.addStream(loggerStream());
+	probotLogger.addStream(loggerStream(false));
 };
 
-const createNewLogger = (name: string, options: LoggerOptions = {}): Logger => {
+interface LoggerOptions {
+	fields?: Record<string, unknown>;
+	streams?: Stream[];
+	level?: LogLevel;
+	stream?: NodeJS.WritableStream;
+	serializers?: Serializers;
+	src?: boolean;
+}
+
+export const getLogger = (name: string, options: LoggerOptions = {}): Logger => {
 	return createLogger(merge<Logger.LoggerOptions, LoggerOptions>({
 		name,
-		stream: LOG_STREAM,
+		streams: [ loggerStream(), loggerStream(false) ],
```

Comment:
Isn't this `getLogger` also used by non frontend middleware code? Then will the `FILTERING_FRONTEND_HTTP_LOGS_MIDDLEWARE_NAME` still make sense?

Reply:
yes :( you pick on my most annoying thing lol, It was inherited from the previous implementation so was originally trying not to change too many things, but that ship sailed so I will have another look if I can do it cleaner!

Reply:
@gxueatlassian I have completely removed the `FILTERING_FRONTEND_HTTP_LOGS_MIDDLEWARE_NAME` const, it has been replaced with a config value. so its much cleaner now its not passing that around everywhere!